### PR TITLE
 implement preemptible_and_regular_instance_templates

### PR DESCRIPTION
**Pull request by namusyaka**
If the option is enabled, this module creates both a regular template
and a template with the preemptible option enabled.
Even if this option and the preemptible option are both set to true,
two types of templates are created because a regular template is created internally.

@morgante @aaron-lane Any thoughts? This is a re-opened PR of #17

### CODE REVIEW

File: modules/instance_template/main.tf [hunk 1]
```diff
@@ -68,6 +68,35 @@ resource "google_compute_instance_template" "tpl" {
...
+    subnetwork         = "${var.subnetwork}"
+    subnetwork_project = "${var.subnetwork_project}"
+  }
+
+  lifecycle {
+    create_before_destroy = "true"
+  }
+
+  scheduling {
+    preemptible       = "${var.preemptible ? "false" : "true"}"
```
Comment (aaron-lane) [inline modules/instance_template/main.tf]: I think we should reconsider how this is implemented. It is confusing that the module has both `var.preemptible` and `var.create_both_preemptible_and_regular`, and the interaction of the two will alter which template is preemptible. I suggest we split this in to two modules, one which creates a preemptible template and one which creates a regular template, with no conditional logic.

Comment (namusyaka) [reply modules/instance_template/main.tf]: As per our PRD, we decided to add a new flag, `create_both_preemptible_and_regular`, for handling the both template case. I'll send you a DM about this.

**Comment by namusyaka** [comment]
@aaron-lane Thanks for your review. It's very helpful. As per our conversation in offline, I have just finished migrating this changes from new flag approach to the new submodule, PTAL.

File: modules/instance_template/main.tf [hunk 2]
```diff
@@ -97,7 +97,9 @@ resource "google_compute_instance_template" "tpl" {
     create_before_destroy = "true"
   }
 
+  // scheduling must have automatic_restart be false when preemptible is true.
   scheduling {
-    preemptible = var.preemptible
+    preemptible       = var.preemptible
+    automatic_restart = ! var.preemptible
```
[the remark this answers is not in the document]
Comment (namusyaka) [reply modules/instance_template/main.tf]: Done. I think there is no problem.
```
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
```

File: modules/preemptible_and_regular_instance_templates/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# preemptible_and_regular_instance_templates
+
+This submodule allows you to create two `google_compute_instance_template`
+resources, that consist of preemptible and regular VMs.
+
+## Usage
+
+See the [simple](examples/preemptible_and_regular_instance_templates/simple) for a usage example.
```
[the remark this answers is not in the document]
Comment (namusyaka) [reply modules/preemptible_and_regular_instance_templates/README.md]: Done: https://github.com/terraform-google-modules/terraform-google-vm/pull/18/commits/74ffff1b2a8e13f61f07b39f64d62bbebcfb4325

File: modules/preemptible_and_regular_instance_templates/outputs.tf [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+output "preemptible_self_link" {
+  value = "${module.preemptible.self_link}"
```
[the remark this answers is not in the document]
Comment (namusyaka) [reply modules/preemptible_and_regular_instance_templates/outputs.tf]: Done: https://github.com/terraform-google-modules/terraform-google-vm/pull/18/commits/61310dbdabeaf0672cef75e0c4a62f94edde2c95

**Comment by namusyaka** [comment]
Upgraded to v0.12.0 syntax: https://github.com/terraform-google-modules/terraform-google-vm/pull/18/commits/88e98815c42879e930a4a3b9084ad2c422eab36b

**Pull request merged**